### PR TITLE
Support relative paths in js modules

### PR DESCRIPTION
**Pull request by letmaik**
Fixes https://github.com/microsoft/CCF/issues/1478.

### CODE REVIEW

File: python/ccf/proposal_generator.py [hunk 1]
```diff
@@ -268,7 +269,12 @@ def set_js_app(app_script_path: str, **kwargs):
 
 @cli_proposal
 def set_module(module_name, module_path, **kwargs):
-    if module_name.endswith(".js"):
+    module_name_ = PurePosixPath(module_name)
+    if not module_name_.is_absolute():
+        raise ValueError("module name must be an absolute path")
+    if any(folder in [".", ".."] for folder in module_name_.parents):
+        raise ValueError("module name must not contain . or .. components")
```
Comment (eddyashton) [inline python/ccf/proposal_generator.py:276]: I may be missing something - doesn't the `is_absolute()` check above forbid this already?

Comment (letmaik) [reply python/ccf/proposal_generator.py:276]: Nope

Comment (letmaik) [reply python/ccf/proposal_generator.py:276]: `/foo/../bar.js` is still an absolute path.

Comment (eddyashton) [reply python/ccf/proposal_generator.py:276]: Right - for some reason I assumed the path construction would normalise.

**Pull request merged**